### PR TITLE
upload logs on e2e failures to help debug

### PR DESCRIPTION
**Pull request by shubhbapna**
I am having trouble debugging macos failures and the only way I can do that is via github actions since I don't have access to a mac machine

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -99,3 +99,8 @@ jobs:
 
       - name: Run tests
         run: ./e2e/test_${{ matrix.test-script }}.sh
+      
+      - name: Upload logs for debugging
+        if: ${{ failure() }}
+        uses: actions/upload-artifact@v4
+        path: e2e-output
```
Comment (dhellmann) [inline .github/workflows/test.yaml:106]: Do we have any data limits we need to be aware of? There might be some big wheels or sdists in that directory.

**Review by dhellmann — COMMENTED** [review]
LGTM, just one question before approving.

File: .github/workflows/test.yaml (hunk 1 above)
Comment (shubhbapna) [reply .github/workflows/test.yaml]: I am not sure i think there is a total limit of 100 GB. Someone experimented and said that greater than 5 GB uploads might start breaking (but i think that is the size of the zip archive): https://github.com/actions/upload-artifact/issues/9

Comment (shubhbapna) [reply .github/workflows/test.yaml]: Could you also the default retention period of artifacts on this repository to be 7 days. It should be somewhere in the repo settings

**Review by dhellmann — APPROVED** [review]

**Comment by dhellmann** [comment]
The retention period is set to 90 days right now. Let's see how that works out, we can adjust if we need to.

**Pull request merged**